src/cli/rustup_mode.rs

```diff
@@ -258,6 +261,15 @@ enum RustupSubcmd {
     },
 }
 
+fn update_toolchain_value_parser(s: &str) -> clap::error::Result<PartialToolchainDesc> {
+    PartialToolchainDesc::from_str(s).map_err(|e| {
+        if s == "self" {
+            info!("if you meant to update rustup itself, use `rustup self update`");
+        }
+        clap::Error::raw(ErrorKind::ValueValidation, e)
```

Comment:
You should throw the original error instead of creating a new one. See https://doc.rust-lang.org/std/result/enum.Result.html#method.inspect_err.